### PR TITLE
Allow multiple digit partition numbers. 

### PR DESCRIPTION
**Pull request by drewmoseley**
Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

### CODE REVIEW

**Comment by drewmoseley** [comment]
This corresponds to changes in https://github.com/mendersoftware/meta-mender/pull/469

File: device.go [hunk 1]
```diff
@@ -137,32 +139,40 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
...
-		return "", errors.New("Error obtaining inactive partition: " + err.Error())
+		return "", "", errors.New("Error obtaining inactive partition: " + err.Error())
 	}
 
 	log.Debugf("Marking inactive partition (%s) as the new boot candidate.", inactivePartition)
 
-	partitionNumber := inactivePartition[len(inactivePartition)-1:]
-	if _, err := strconv.Atoi(partitionNumber); err != nil {
-		return "", errors.New("Invalid inactive partition: " + inactivePartition)
+	partitionSlice := strings.SplitAfter(inactivePartition, "p")
```
Comment (kacf) [inline device.go]: Repeating what is in https://github.com/mendersoftware/meta-mender/pull/469:

This won't work for `/dev/sda2` type of partitions, which is likely to be common on larger boards, such as Intel. I think you will have to count digits backwards from the end of the string.

Suggestion (untested):

```
partitionNumberDecStr := inactivePartition[len(strings.TrimRight(inactivePartition, "0123456789")):]
```

which gives you the final digits.

**Comment by kacf** [comment]
Again, I don't know what's going on with the tests. Did a Fabric update suddenly break SSH connectivity? I'll be looking into this, currently *all* our tests are failing.

**Comment by kacf** [comment]
I think it may have been fixed upstream. Rerunning.

Raspberry Pi is known to be broken ATM, so just ignore that one.

**Comment by kacf** [comment]
I think there is definitely *something* wrong with this code, because your PR is the only one exhibiting these test problems. Maybe fix the issue I pointed out in my review first and we can take it from there.

**Comment by drewmoseley** [comment]
@kacf I updated this and tested it as far as I was able.  I only have the one board that uses multiple digit partition numbers so it's possible there are other cases that may need some treatment but the Dragonboard I have on the Morty branch works.

**Comment by drewmoseley** [comment]
> 	Error:		Max difference between 2018-02-06 00:10:14.063482976 +0000 UTC and 2018-02-06 00:10:14.059387126 +0000 UTC allowed is 1ms, but difference was 4.09585ms

Could it really be that this simple parsing change is taking 3+ ms?

**Comment by kacf** [comment]
Sigh, the tests are such an unpredictable bunch. I have restarted it yet again.

**Comment by kacf** [comment]
So this one looks like it could be legit, since it's all client failures: https://s3-eu-west-1.amazonaws.com/mender-testing-reports/integration-reports/1034/report.html

However, I will rerun once more just in case.

**Comment by drewmoseley** [comment]
It seems like strange errors to have been caused by this particular change.  Any idea how that could happen?

**Comment by kacf** [comment]
Oh wait, I think I know. Can you rebase your changes on top of the latest mender/master? I think this may be because you're missing Ole's reboot hardening which was merged last week.

**Comment by drewmoseley** [comment]
Done.  Monitoring CI.  Fingers crossed.

**Comment by kacf** [comment]
Alright, all current failures are expected due to the recent Jenkins difficulty. Merging.

**Comment by drewmoseley** [comment]
@kacf what branches will this be in?  Will it get backported or only stay in master and then the next release?

**Comment by kacf** [comment]
I think this can be cherry-picked. Pretty safe, and a bug fix. I opened these two:

* #273
* #274

**Pull request merged**